homeassistant/components/solax/sensor.py

```diff
@@ -151,7 +151,7 @@ async def async_refresh(self, now=None):
             raise PlatformNotReady from err
         data = api_response.data
         for sensor in self.sensors:
-            if sensor.key in data:
+            if sensor.key in data and sensor.enabled:
```

Comment:
The correct solution is to use our `DataUpdateCoordinator` helper to fetch the data and update the entities.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

The problem that this PR is addressing is not the only problem with the current design. I'm hesistant accepting another workaround which adds code that isn't needed if we would use the correct design.

Reply:
I understand. Unfortunately that's a bigger change than I can take on at this moment, perhaps another time in a new PR. Thanks for the review. 👍